src/win/util.c

```diff
@@ -484,7 +484,7 @@ uint64_t uv__hrtime(double scale) {
    * performance counter interval, integer math could cause this computation
    * to overflow. Therefore we resort to floating point math.
    */
-  return (uint64_t) ((double) counter.QuadPart * hrtime_interval_ * scale);
+  return (uint64_t) (counter.QuadPart * (uint64_t) scale / hrtime_interval_);
```

Comment:
> But using integer math is the wrong approach IMO, and can introduce overflow and rounding errors.

I'm not convinced that the timers-first-fire failure is caused by rounding errors either.
I disagree that there's something wrong in principle with using integer math though; rounding errors can be avoided by multiplying *before* dividing, while overflow can be curbed by using a big enough type.